Modules/Core/SpatialObjects/include/itkSpatialObjectPoint.h

```diff
@@ -158,7 +158,7 @@ class ITK_TEMPLATE_EXPORT SpatialObjectPoint
   /** Color of the point */
   ColorType m_Color;
 
-  typename SpatialObjectType::Pointer m_SpatialObject;
+  WeakPointer<SpatialObjectType> m_SpatialObject;
```

Comment:
Perhaps add a comment "//The PointSpatialObject keeps a reference to its parent spatial object for its spatial context, so use a WeakPointer here to avoid a memory leak."